File: src/main/java/com/topface/topface/requests/DeleteVisitorsRequest.java
package com.topface.topface.requests;

import android.content.Context;

import java.util.List;

public class DeleteVisitorsRequest extends DeleteAbstractUsersRequest {
    private static final String SERVICE = "visitor.delete";

    public DeleteVisitorsRequest(List<String> ids, Context context) {
        super(ids, context);
    }

    @Override
    public String getServiceName() {
        return SERVICE;
    }

    @Override
    protected String getKeyForItems() {
        return "ids";
    }
}

File: src/main/java/com/topface/topface/api/requests/DeleteVisitorsRequest.kt
package com.topface.topface.api.requests

import com.google.gson.JsonObject
import com.topface.topface.api.responses.Completed
import com.topface.topface.api.stringListToJsonArray
import java.util.*

class DeleteVisitorsRequest(val userIds: ArrayList<String>) : BaseScruffyRequest<Completed>() {

    override fun getMethod() = "visitor.delete"

    override fun createJson(json: JsonObject) = with(json) {
        add("ids", userIds.stringListToJsonArray())
    }

    override fun getResponseClass(): Class<Completed> = Completed::class.java
}